test/integration/shoots/applications/shoot_app_test.go

```diff
@@ -0,0 +1,307 @@
+		shootTestOperations.ShootGardenerTest.Logger.Infof("Shoot Kubeconfig downloaded successfully to %s", *downloadPath)
+	}, DownloadKubeconfigTimeout)
+
+	CIt("should deploy guestbook app successfully", func(ctx context.Context) {
+		ctx = context.WithValue(ctx, "name", "guestbook app")
+		helmDeployNamespace := metav1.NamespaceSystem
+		defer func() {
+			shootTestOperations.ShootGardenerTest.Logger.Info("Cleaning up resources")
+
+			cleanupGuestbook := func() {
+				err := shootTestOperations.K8sShootClient.DeleteIngress(helmDeployNamespace, "guestbook")
+				Expect(err).NotTo(HaveOccurred())
+
+				err = shootTestOperations.K8sShootClient.DeleteDeployment(helmDeployNamespace, "guestbook")
+				Expect(err).NotTo(HaveOccurred())
+
+				err = shootTestOperations.K8sShootClient.DeleteService(helmDeployNamespace, "guestbook")
+				Expect(err).NotTo(HaveOccurred())
+			}
+
+			//TODO: an enhancement could be to implement helm purge functionality into the testing framework
+			cleanupRedis := func() {
+				err := shootTestOperations.K8sShootClient.DeleteDeployment(helmDeployNamespace, "redis-slave")
+				Expect(err).NotTo(HaveOccurred())
+
+				err = shootTestOperations.K8sShootClient.DeleteService(helmDeployNamespace, "redis-master")
+				Expect(err).NotTo(HaveOccurred())
+
+				err = shootTestOperations.K8sShootClient.DeleteService(helmDeployNamespace, "redis-slave")
+				Expect(err).NotTo(HaveOccurred())
+
+				err = shootTestOperations.K8sShootClient.DeleteStatefulSet(helmDeployNamespace, "redis-master")
+			}
+			cleanupGuestbook()
+			cleanupRedis()
+
+			shootTestOperations.ShootGardenerTest.Logger.Info("redis and the guestbook app have been cleaned up!")
+		}()
+
+		helm := Helm(ResourcesDir)
+		err := shootTestOperations.EnsureDirectories(helm)
+		Expect(err).NotTo(HaveOccurred())
+
+		err = shootTestOperations.DownloadAndDeployHelmChart(ctx, helm, helmDeployNamespace, "stable/redis")
+		Expect(err).NotTo(HaveOccurred())
+
+		err = shootTestOperations.WaitUntilPodIsRunning(ctx, "redis-master-0", helmDeployNamespace, shootTestOperations.K8sShootClient)
+		Expect(err).NotTo(HaveOccurred())
+
+		redisSlaveLabels := map[string]string{
+			"app":  "redis",
+			"role": "slave",
+		}
+
+		err = shootTestOperations.WaitUntilPodWithLabelsIsReady(ctx, redisSlaveLabels, helmDeployNamespace, shootTestOperations.K8sShootClient)
+		Expect(err).NotTo(HaveOccurred())
+
+		guestBookParams := struct {
+			HelmDeployNamespace string
+			ShootDNSHost        string
+		}{
+			helmDeployNamespace,
+			fmt.Sprintf("guestbook.ingress.%s", *shootTestOperations.Shoot.Info.Spec.DNS.Domain),
+		}
+
+		// Parse the guestbook app template
+		var writer bytes.Buffer
+		err = guestBooktpl.Execute(&writer, guestBookParams)
+		Expect(err).NotTo(HaveOccurred())
+
+		// Apply the guestbook app resources to shoot
+
+		manifestReader := kubernetes.NewManifestReader(writer.Bytes())
+		err = shootTestOperations.K8sShootClient.Applier().ApplyManifest(context.TODO(), manifestReader)
```

Comment:
Don't use `context.TODO()`.

Reply:
context is not adopted in this realm. When it is, this can be changed.

Reply:
it is indeed 'adopted in this realm'. Your function already has a proper context, so use it.

Reply:
As discussed, not adopted in this realm :)